internal/api/internal/client.go

```diff
@@ -0,0 +1,44 @@
+package internal
+
+import (
+	"errors"
+
+	"github.com/deepmap/oapi-codegen/pkg/securityprovider"
+)
+
+// TODO(andrew): most of this is boilerplate that should be generated
+
+const serverBase = "/api/internal"
+
+type ClientConfig struct {
+	Server string
+	Base   string
+	Token  string
+}
+
+type APIClient struct {
+	client *ClientWithResponses
+}
+
+func CreateClient(config ClientConfig) (*APIClient, error) {
+	if config.Server == "" {
+		return nil, errors.New("must specify a server value")
+	}
+	clientOptions := []ClientOption{}
+
+	if config.Token != "" {
+		apiKeyProvider, _ := securityprovider.NewSecurityProviderApiKey("header", "X-API-Gateway-Token", config.Token)
```

Comment:
I can leave a comment about this ignored error, but it's ignored because the initialization method is just kind of silly:

https://github.com/deepmap/oapi-codegen/blob/9c600dddea334bff2fa527dd14af1aa2900917a8/pkg/securityprovider/securityprovider.go#L72-L98

if the first parameter is `"header"`, `"query"` or `"cookie"` it'll never return an error.